src/go/k8s/pkg/networking/address_types.go

```diff
@@ -0,0 +1,57 @@
+package networking
+
+import (
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const defaultType = corev1.NodeExternalIP
+
+var typePriority = []corev1.NodeAddressType{
+	corev1.NodeExternalDNS,
+	corev1.NodeExternalIP,
+	corev1.NodeInternalDNS,
+	corev1.NodeInternalIP,
+	corev1.NodeHostName,
+}
+
+// GetPreferredAddress returns the preferred node address if
+// it is available and a preferred type is provided. Otherwise,
+// it loops over the address types by priority and returns the
+// first one that is available.
+func GetPreferredAddress(
+	node *corev1.Node, preferredType corev1.NodeAddressType,
+) string {
+	if node == nil {
+		return ""
+	}
+	if preferredType == "" {
+		preferredType = defaultType
+	}
+	// Return the address of the preferred type, if found
+	if s := findAddressType(node, preferredType); s != "" {
+		log.Println("address:", s, "type:", preferredType)
```

Comment:
nit: this should use controller logger or not write to log (also other cases in this file)

Reply:
yeah, those are removed